client/signup/config/steps.js

```diff
@@ -85,17 +86,8 @@ export default {
 		providesToken: true,
 		providesDependencies: [ 'bearer_token', 'username' ],
 		unstorableDependencies: [ 'bearer_token' ],
-	},
-
-	'user-social': {
-		stepName: 'user-social',
-		apiRequestFunction: stepActions.createAccount,
-		providesToken: true,
-		providesDependencies: [ 'bearer_token', 'username' ],
-		unstorableDependencies: [ 'bearer_token' ],
 		props: {
-			headerText: i18n.translate( 'Create your account.' ),
-			isSocialSignupEnabled: true
+			isSocialSignupEnabled: ! config.isEnabled( 'desktop' )
```

Comment:
What about using the `signup/social` flag instead? [As far I can see](https://github.com/Automattic/wp-calypso/tree/6ab1e78805500629727586eb25280316fda9162f/config) it's enabled for all environments except `desktop`.